### DISCUSSION TITLE
PackageReference by Variable (Nuget)

### DISCUSSION THREAD

**Discussion by Nico-VanHaaster** [renovatebot/renovate#18976] [Request Help]
Hello,

First off great product! We have just started implementing renovate to our mono-repo (100s of solutions) and one common pattern we have adopted across all the solutions has been grouping our dependencies from common sources using variables such as the example below.

```
<Project>
  <PropertyGroup>
    <ElsaVersion>2.8.2</ElsaVersion>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Update="Elsa" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.Activities.Console" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.Activities.RabbitMq" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.Activities.Http" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.Core" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.DataMasking.Core" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.Server.Api" Version="$(ElsaVersion)" />
    <PackageReference Update="Elsa.WorkflowTesting.Api" Version="$(ElsaVersion)" />
  </ItemGroup>
</Project>
```
Where we have defined a top level variable for packages that typically are released at the same time. This reduces the requirements of updating each package version manually. I cant find any documentation regarding support for scenario, also the debug output is not showing any details regarding trying to upgrade these packages.

Do you think this should be a supported scenario, or by design should we move back to specifying these versions manually?

**Comment by viceice** [comment]
it's currently not supported, but we support variables in maven and Gradle, so we can probably do this in nuget too. but that won't have high priority.

so your only workaround to use renovate for now is to not use variables and let renovate update and group as required.

**Reply by Nico-VanHaaster** [reply]
Thanks for the very quick response! I suspected so, although this is a one-time hit for us to upgrade all our projects, that being said if we adopt renovate correctly then realistically, the manual updates will be limited.
